### DISCUSSION TITLE
[Feature-Request]: Azure related data sources should have managed identity based Authentication

### DISCUSSION THREAD

**Discussion by cprasad-1512** [grafana/grafana#43552] [Data Sources]
**Current Behavior:**

Right now, we can see **Managed Identity** Authentication type in **Azure Monitor** data source only.
And, this option can be **enable** from Grafana **config** file using this keyword ```managed_identity_enabled = true``` under **[azure]** section.
As you can see below:

![image](https://user-images.githubusercontent.com/80390752/147549342-679317f9-ca80-4d0f-be03-94d8b5827169.png)

**Request:**
If it's possible, can we have **Managed Identity** Authentication in other **Azure** or **Microsoft** data Sources like Mirosoft SQL Server, Azure Data Explorer, etc.

**Benefits for this:**

1. Using this, App Client ID and Client Secret will not expose on UI.
2. Security measure will increase
3. Using this, we can work with Grafana more efficiently

Thanks!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
